mathesar_ui/src/stores/abstract-types/operations/filtering.ts

```diff
@@ -218,13 +219,14 @@ export const filterDefinitionMap = getFilterDefinitionMap();
 
 export function getFiltersForAbstractType(
   categoryIdentifier: AbstractTypeCategoryIdentifier,
+  isFK: FkConstraint
 ): Map<AbstractTypeFilterDefinition['id'], AbstractTypeFilterDefinition> {
   const allowedFiltersMap: Map<
     AbstractTypeFilterDefinition['id'],
     AbstractTypeFilterDefinition
   > = new Map();
 
-  filterDefinitionMap.get(categoryIdentifier)?.forEach((filter) => {
+  filterDefinitionMap.get((isFK !== undefined ? "boolean"  : categoryIdentifier))?.forEach((filter) => {
```

Comment:
Is it better to have a separate method in this file for getting filters which are only for equality checks (to be used with PKs and FKs), which include the `equal` and `empty`. That method should then be called to get the filters, instead of getting the filters of boolean.

Reply:
Thank you so much for the guidance. I will start trying to implement these changes. I am still quite new to JS so it may take some time. Thank you again for the help, I really appreciate it.